### PR TITLE
Add widget test for portrait mode.

### LINKED ISSUE #28: Add tests for dialogs in portrait mode and Assertion checks.
In order to increase the codecov report, we need to cover all the possible test cases.

- [ ] - NetworkGiffyDialog test (Portrait mode)
- [ ] - AssetGiffyDialog test (Portrait mode)
- [ ] - FlareGiffyDialog test (Portrait mode)
- [ ] - Test for covering all the possible asserts.

### PR DESCRIPTION
**Pull request by xsahil03x**
Fix #28 
- [x] - NetworkGiffyDialog test (Portrait mode)
- [x] - AssetGiffyDialog test (Portrait mode)
- [x] - FlareGiffyDialog test (Portrait mode)
- [x] - Test for covering all the possible assertions.

### CODE REVIEW

**Comment by xsahil03x** [comment]
@boyan01 can you help me find out why my tests are not covering the `Orientation.portrait` part of [CodeCov](https://codecov.io/gh/xsahil03x/giffy_dialog)

File: test/giffy_dialog_test.dart [hunk 1]
```diff
@@ -1,39 +1,136 @@
...
+
+        expect(find.byKey(keys[5]), findsOneWidget);
+      });
+    });
+
+    testWidgets('Check Landscape Dialog test', (WidgetTester tester) async {
+      provideMockedNetworkImages(() async {
+        final TestWidgetsFlutterBinding binding =
+            TestWidgetsFlutterBinding.ensureInitialized();
+        await binding.setSurfaceSize(Size(LANDSCAPE_WIDTH, LANDSCAPE_HEIGHT));
```
Comment (boyan01) [inline test/giffy_dialog_test.dart]:
```suggestion
        binding.window.physicalSizeTestValue = Size(PORTRAIT_WIDTH, PORTRAIT_HEIGHT);
```
MediaQuery use `window.physicalSize` to obtain the device size

Comment (boyan01) [reply test/giffy_dialog_test.dart]: I think that is the reason which cause you test could not covering the `Orientation.portrait` case.

Comment (xsahil03x) [reply test/giffy_dialog_test.dart]: @boyan01 It didn't work, the tests are failing, can you please have a look ?

`Expected: exactly one matching node in the widget tree
  Actual: ?:<zero widgets with key [<'FlareDialog'>] (ignoring offstage widgets)>
   Which: means none were found but one was expected
`

Comment (boyan01) [reply test/giffy_dialog_test.dart]: increase the screen size can fix it
```dart
const double PORTRAIT_WIDTH = 1800.0;
const double PORTRAIT_HEIGHT = 2400.0;
const double LANDSCAPE_WIDTH = 2400.0;
const double LANDSCAPE_HEIGHT = 1800.0;
```

Comment (xsahil03x) [reply test/giffy_dialog_test.dart]: @boyan01 The test got passed, but it still didn't cover the `Orientation.portrait` case.

Comment (xsahil03x) [reply test/giffy_dialog_test.dart]: > MediaQuery use `window.physicalSize` to obtain the device size

Do we really need to change this size parameter here? As flutter runs the test in landscape mode by default.

Comment (boyan01) [reply test/giffy_dialog_test.dart]: > > MediaQuery use `window.physicalSize` to obtain the device size
> 
> Do we really need to change this size parameter here? As flutter runs the test in landscape mode by default.

sorry, my bad.  I just paste it into wrong line, it maybe only need change the size when run `Check Portrait Dialog test`

Comment (xsahil03x) [reply test/giffy_dialog_test.dart]: Hey, it got fixed. Thanks again for your help.

[link](https://codecov.io/gh/xsahil03x/giffy_dialog/branch/increase_codecov)

**Pull request merged**